feat: Document dynamic sampling (#2954)
getsentry/sentry-docs: src/docs/product/data-management-settings/dynamic-sampling/getting-started.mdx

---
title: Getting Started
sidebar_order: 20

> description: "Quick introduction into setting up dynamic sampling."

Reviewer: ```suggestion
description: "Learn about setting up dynamic sampling."
```

---

Getting started with dynamic sampling is simple, in this section we will go through setting up dynamic sampling for a project.

<Alert level="info" title="Note">

You need to have organization owner or manager role in order to edit project settings.

</Alert>

> 

Reviewer: ```suggestion

## Basic Rule Setup

```

We want to create a rule that applies only to our current release (e.g. release 1.2.1) and we would like to keep all errors from it.

Select 'Add error rule', then select 'Match custom conditions' , select Category "Releases" and Matches: 1.2.1 ; set the sampling rate at 100%.

![release-rule](release-rule.png)

Finally we would like to create a rule that applies to all releases which keeps only 10% of the errors.

Select 'Add error rule', then select "Apply to all" and then set a sampling rate of 10%.

Now all errors originating in release 1.2.1 are going to be kept while errors for other releases will be sampled at a rate of 10%.

<Alert level="info" title="Order of Precedence">

Rules evaluate top to bottom, with the first matching rule determining the sample rate. It is important that the rule for the specific release comes before the rule for all events. If created the other way around, you can reorder by draging the rules in the list.

</Alert>

![rules-example](rules-example.png)

> 

Reviewer: ```suggestion
## Further Configuration

```

> There is a lot more to dynamic sampling than what we have seen in this example, continue reading to find out how to take advantage of all the features offered by dynamic sampling.

Reviewer: ```suggestion
There is a lot more to dynamic sampling than what we have seen in this example. In project settings under "Filters & Sampling",  you can find two lists one for error rules and one for transaction rules. The transaction rules list contains both rules for independent transactions and rules for transaction traces.

Regardless of the type of rule the basic structure of the rule is the same:

* has a type that determines the event type to which it applies (Error or Transaction).
* has 0 or more conditions
* has a sampling rate (in percentages)

If a rule contains multiple conditions the conditions will have to be met simultaneously (`and` logical operation) in
order to have the rule applied.

A condition may contain multiple values to be matched, for the condition to be matched any of values must match
(`or` logical operation).
```
